pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -38,16 +38,44 @@ let
   mkUniqueOutPaths = pkgs: lib.unique
     (map (p: p.outPath) (builtins.filter lib.isDerivation pkgs));
 
-in (buildEnv {
   name = "texlive-${extraName}-${bin.texliveYear}${extraVersion}";
 
-  extraPrefix = "/share/texmf";
+  texmf = (buildEnv {
+    name = "${name}-texmf";
+
+    paths = pkgList.nonbin;
+
+    nativeBuildInputs = [ perl bin.core.out ];
+
+    postBuild = # generate ls-R database
+    ''
+      perl -I "${bin.core.out}/share/texmf-dist/scripts/texlive" \
+        -- "$out/scripts/texlive/mktexlsr.pl" --sort "$out"
+    '';
+  }).overrideAttrs (_: { allowSubstitutes = true; });
```

Comment:
Does buildEnv really not take allowSubstitutes? If so we should fix this at buildEnv level.

Reply:
That would be great. Right now `buildEnv` only accepts a fixed list of attributes. We could add e.g. `extraArgs` to `buildEnv`?

Reply:
or better we accept the standard ones mkDerivation does and that make sense, too. extraArgs is usually only an escape hatch.

Reply:
@SuperSandro2000 do you mind if I start a separate `buildEnv` PR for this? In the meanwhile, if you don't spot anything suspicious and @veprbl is also happy, we merge this one.

I am mindful of the TeX Live 2022 final release approaching and it would be best not to have pending TeX Live infrastructure changes while that happens. I assume that a change to `buildEnv` will take some time, even if tiny, given how central it is to Nixpkgs.